pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -344,6 +314,7 @@ func deleteMachineConfig(r *ReconcileComplianceRemediation, name string, logger
 		logger.Info("MC to be deleted was already deleted")
 		return nil
 	} else if err != nil {
+		// delete error should be retried
```

Comment:
Thank you for keeping and adding the comments. It shows that someone thought about the logic and it's not just copy-paste code.